internal/cephfs/clone.go

```diff
@@ -130,6 +131,9 @@ func createCloneFromSubvolume(ctx context.Context, volID, cloneID volumeID, volO
 			util.ErrorLog(ctx, "failed to delete snapshot %s %v", snapshotID, err)
 			return err
 		}
+	case cephFSCloneError:
+		util.ErrorLog(ctx, "fetching the clone state returned an error %v", cloneID)
+		return ErrCloneState
```

Comment:
```suggestion
	case cephFSCloneError:
		return cloneErr
```
We need to return the actual error.  
There is more descriptive Errorlog in function https://github.com/ceph/ceph-csi/blob/fb365005da1be708cb04502e6217e97de23efe90/internal/cephfs/clone.go#L218. 
Move this case to the top of switch statement and remove error check at line 98. 
This change should come within `gocyclo` limit, so remove nolinter for same too.